contracts/BasicPolyptychRandomizerV0.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Creatd By: Art Blocks Inc.
+
+pragma solidity 0.8.17;
+
+import "./interfaces/0.8.x/IBasicPolyptychRandomizerV0.sol";
+import "./interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "./interfaces/0.8.x/IFilteredMinterHolderV1.sol";
+
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+
+/**
+ * @title Basic Polytych Randomizer contract that enables the copying of token hash seeds.
+ * @notice This contract requires several configuration options to be set properly before
+ * it enables the copying of token hashes from an original token (the first panel of a polyptych)
+ * to subsequent diptych, triptych, or polyptych panels.
+ *
+ * Requirements to use the polyptych randomizer:
+ * - Core contract must be an instance of `GenArt721CoreV3_Engine`
+ * - The initial project used as the first panel must be minted with a completely random token hash
+ * - An `MinterPolyptychV0` contract or similar must be deployed and configured as the `hashSeedSetterContract`
+ * - The `projectMaximumHashSeedCopies` should be configured for each project
+ *
+ * Once the contract is configured, it will enable newly-minted tokens to use a copy of the hash
+ * seed from a previously-minted token.
+ */
+contract BasicPolyptychRandomizerV0 is IBasicPolyptychRandomizerV0, Ownable {
+    // The core contract that may interact with this randomizer contract.
+    IGenArt721CoreContractV3 public genArt721Core;
+
+    // Used to obtain the project ID from the token ID
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    // Mapping of token IDs to hash seeds
+    mapping(uint256 => bytes12) public polyptychHashSeeds;
+
+    // Mapping of the number of times a hash seed has been used
+    mapping(bytes12 => uint) public hashSeedUsageCounter;
+
+    // Configures the maximum number of hash seed copies that can exist per project (default zero)
+    mapping(uint256 => uint) public projectMaximumHashSeedCopies;
```

Comment:
Instead of being a boolean, this keeps track of the number of times we allow a project's token hash to be copied.

This limit is set per-project to allow two different use cases:
1. If polyptych panels are all part of the same project, the artist can increment this maximum limit (default zero) to allow the next panel to be minted on the same project ID
2. If polyptych panels are in a new project, the new project should have a limit that is equal to the previous number of panels